sktime/forecasting/base/adapters/_darts.py

```diff
@@ -0,0 +1,281 @@
+# copyright: sktime developers, BSD-3-Clause License (see LICENSE file)
+"""Implements adapter for Darts models."""
+import abc
+from typing import List, Optional, Union
+
+import pandas as pd
+
+from sktime.forecasting.base import BaseForecaster, ForecastingHorizon
+
+__author__ = ["yarnabrina", "fnhirwa"]
+
+
+class _DartsAdapter(BaseForecaster):
+    """Base adapter class for Darts models.
+
+    Parameters
+    ----------
+    past_covariates : Optional[List[str]], optional
+        column names in ``X`` which are known only for historical data, by default None
+    num_samples : Optional[int], optional
+        Number of times a prediction is sampled from a probabilistic model, by default
+        1000
+
+    Notes
+    -----
+    If unspecified, all columns will be assumed to be known during prediction duration.
+    """
+
+    _tags = {
+        # packaging info
+        # --------------
+        "authors": ["yarnabrina", "fnhirwa"],
+        "maintainers": ["yarnabrina", "fnhirwa"],
+        "python_version": ">=3.9",
+        "python_dependencies": ["u8darts"],
+        "python_dependencies_alias": {"u8darts": "darts"},
+        # estimator type
+        # --------------
+        "y_inner_mtype": "pd.DataFrame",
+        "X_inner_mtype": "pd.DataFrame",
+        "requires-fh-in-fit": False,
+        "enforce_index_type": pd.DatetimeIndex,
+        "handles-missing-data": False,
+        "capability:insample": False,
+    }
+
+    def __init__(
+        self: "_DartsAdapter",
+        past_covariates: Optional[Union[List[str], None]] = None,
+        num_samples: Optional[int] = 1000,
+    ) -> None:
+        if not isinstance(past_covariates, list) and past_covariates is not None:
+            raise TypeError(
+                f"Expected past_covariates to be a list, found {type(past_covariates)}."
+            )
+        self.past_covariates = past_covariates
+        if not isinstance(num_samples, int):
```

Comment:
Does it mean that users can not pass `None`? In the `__init__` signature, `None` is allowed since you used `Optional`, which means `Union[..., None]` (ref. https://docs.python.org/3/library/typing.html#typing.Optional).